infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutAssetDao.kt

```diff
@@ -133,26 +121,28 @@ abstract class LayoutAssetDao<T : LayoutAsset<T>>(
 
     private val allPublicationVersionsSql = """
         select
-          coalesce(official_row_id, id) as official_id,
+          coalesce(official_row_id, design_row_id, id) as official_id,
           id as row_id,
           version as row_version
         from ${table.fullName}
-        where draft = true
+        where draft and design_id is not distinct from :design_id
```

Comment:
En ole itse juuri nähnyt tätä muotoa käytettävän aiemmin. Onkos tällä mitään käytännön eroa siihen, että tämä olisi kirjoitettu muodossa `and design_id = :design_id`?

Reply:
On: `design_id = :design_id` SQL:ssä null, jos kumpikaan arvo on null, ja siis myös jos molemmat on null. Nullius sitten kulkee myös kaikkien `and`ien ja `or`ien läpi, ja lopulta `where` tulkitsee nullit falseiksi. Ts. jos tässä olisi "design_id = :design_id", niin normaalitapaus että rivi on main-haarassa ja on pyydetty main-haaraa, ei menisikään enää läpi.